src/OnlineVotingS.Application/Services/ImplService/CampaignService.cs

```diff
@@ -12,71 +13,176 @@
                 throw new KeyNotFoundException($"Campaign with ID {campaignId} not found.");
             }
 
             await _campaignRepository.DeleteAsync(campaignId);
+
+            _logger.LogInformation("Campaign with ID {CampaignId} deleted successfully.", campaignId);
             return true;
         }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "An error occurred while deleting the campaign with ID {CampaignId}.", campaignId);
+            throw;
+        }
+    }
 
-        public async Task<Campaign> GetCampaignByIdAsync(int campaignId)
+    public async Task<Campaign> GetCampaignByIdAsync(int campaignId)
+    {
+        _logger.LogInformation("Fetching campaign with ID {CampaignId}.", campaignId);
+
+        try
         {
-            return await _campaignRepository.GetByIdAsync(campaignId);
+            var campaign = await _campaignRepository.GetByIdAsync(campaignId);
+            if (campaign == null)
+            {
+                _logger.LogWarning("Campaign with ID {CampaignId} not found.", campaignId);
+                throw new KeyNotFoundException($"Campaign with ID {campaignId} not found.");
+            }
+
+            return campaign;
         }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "An error occurred while fetching the campaign with ID {CampaignId}.", campaignId);
+            throw;
+        }
+    }
 
-        public async Task<IEnumerable<Campaign>> GetAllCampaignsAsync()
+    public async Task<IEnumerable<Campaign>> GetAllCampaignsAsync()
+    {
+        _logger.LogInformation("Fetching all campaigns.");
+
+        try
+        {
+            var campaigns = await _campaignRepository.GetAllAsync();
+            _logger.LogInformation("Fetched {Count} campaigns.", campaigns.Count());
+            return campaigns;
+        }
+        catch (Exception ex)
         {
-            return await _campaignRepository.GetAllAsync();
+            _logger.LogError(ex, "An error occurred while fetching all campaigns.");
+            throw;
         }
+    }
+
+    public async Task<IEnumerable<Campaign>> GetCampaignsByElectionIdAsync(int electionId)
+    {
+        _logger.LogInformation("Fetching campaigns for election ID {ElectionId}.", electionId);
 
-        public async Task<IEnumerable<Campaign>> GetCampaignsByElectionIdAsync(int electionId)
+        try
         {
-            return await _campaignRepository.GetByElectionIdAsync(electionId);
+            var campaigns = await _campaignRepository.GetByElectionIdAsync(electionId);
+            _logger.LogInformation("Fetched {Count} campaigns for election ID {ElectionId}.", campaigns.Count(), electionId);
+            return campaigns;
         }
+        catch (Exception ex)
+        {
+            _logger.LogError(ex, "An error occurred while fetching campaigns for election ID {ElectionId}.", electionId);
```

Comment:
follow please if it works the same guidline as I mentonied here https://github.com/ArianitSGashi/OnlineVotingS/pull/45/files#r1723195909